src/Middleware/UriReplacementHandler.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Kiota.Abstractions;
+using Microsoft.Kiota.Http.HttpClientLibrary.Extensions;
+
+namespace Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+
+/// <summary>
+/// Interface for making URI replacements.
+/// </summary>
+public interface IUriReplacementHandlerOption : IRequestOption
+{
+    /// <summary>
+    /// Check if URI replacement is enabled for the option.
+    /// </summary>
+    /// <returns>true if replacement is enabled or false otherwise.</returns>
+    bool IsEnabled();
+
+    /// <summary>
+    /// Accepts a URI and returns a new URI with all replacements applied.
+    /// </summary>
+    /// <param name="original">The URI to apply replacements to</param>
+    /// <returns>A new URI with all replacements applied.</returns>
+    Uri? Replace(Uri? original);
+}
+
+/// <summary>
+/// Url replacement options.
+/// </summary>
+public class UriReplacementHandlerOption : IUriReplacementHandlerOption
```

Comment:
For consistency with the rest of the project middleware options, I would suggest moving this class to the Options folder/namespace.